test/models/patient_notification_eligibility_test.rb

```diff
@@ -0,0 +1,650 @@
+      past_monitoring_period_dependent_test(patient)
+    end
+  end
+
+  test 'non-HoH, eligible flows' do
+    ['E-mailed Web Link', 'SMS Texted Weblink', 'Telephone call', 'SMS Text-message'].each do |preferred_contact_method|
+      [
+        { isolation: true },
+        { continuous_exposure: true },
+        { last_date_of_exposure: nil, created_at: default_days_ago(5) },
+        { last_date_of_exposure: default_days_ago(5), created_at: default_days_ago(5) },
+        { last_date_of_exposure: default_days_ago(11), created_at: default_days_ago(20) }
+      ].each do |workflow_params|
+        patient = create(
+          :patient,
+          {
+            preferred_contact_method: preferred_contact_method,
+            pause_notifications: false,
+            purged: false,
+            monitoring: true,
+            closed_at: nil
+          }.merge(workflow_params)
+        )
+        assert_eligible(patient)
+        # Creating an assessment from yesterday SHOULD NOT affect eligibility
+        create(
+          :assessment,
+          patient: patient,
+          symptomatic: false,
+          created_at: default_days_ago(1).end_of_day
+        )
+        assert_eligible(patient)
+        # Creating an assessment from today SHOULD affect eligibility
+        create(
+          :assessment,
+          patient: patient,
+          symptomatic: false,
+          created_at: default_days_ago(0).beginning_of_day
+        )
+        assert_ineligible(patient)
+      end
+    end
+  end
+
+  test 'non-HoH, workflow-shared, non-eligible flows' do
+    ['E-mailed Web Link', 'SMS Texted Weblink', 'Telephone call', 'SMS Text-message'].each do |preferred_contact_method|
+      [
+        { isolation: true },
+        { continuous_exposure: true },
+        { last_date_of_exposure: nil, created_at: default_days_ago(5) },
+        { last_date_of_exposure: default_days_ago(5), created_at: default_days_ago(5) },
+        { last_date_of_exposure: default_days_ago(11), created_at: default_days_ago(20) }
+      ].each do |workflow_params|
+        [
+          { pause_notifications: true },
+          { monitoring: false, closed_at: default_days_ago(1) },
+          { last_assessment_reminder_sent: default_days_ago(0) - 11.hours },
+          { preferred_contact_method: 'Unknown' },
+          { preferred_contact_method: 'Opt-out' },
+          { preferred_contact_method: '' },
+          { preferred_contact_method: nil }
+        ].each do |ineligible_params|
+          patient_args = {
+            preferred_contact_method: preferred_contact_method,
+            pause_notifications: false,
+            purged: false,
+            monitoring: true,
+            closed_at: nil
+          }.merge(workflow_params).merge(ineligible_params)
+          patient = create(:patient, patient_args)
+          assert_ineligible(patient)
+        end
+      end
+    end
+  end
+
+  test 'non-HoH, special exposure workflow, non-eligible flows' do
```

Comment:
These test names aren't the clearest - a comment would help. Not sure what "special exposure workflow" refers to or should be testing.

Reply:
Changing some test names and adding more test comments